sound/soc/sdca/sdca_interrupts.c

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * SDCA IRQ handler library
+ *
+ * Copyright (C) 2025 Cirrus Logic, Inc. and
+ *                         Cirrus Logic International Semiconductor Ltd.
+ *
+ * The MIPI SDCA specification is available for public downloads at
+ * https://www.mipi.org/mipi-sdca-v1-0-download
+ */
+#include <linux/device.h>
+#include <linux/soundwire/sdw.h>
+#include <linux/soundwire/sdw_registers.h>
+#include <sound/sdca.h>
+#include <linux/regmap.h>
+#include <linux/interrupt.h>
+#include <linux/workqueue.h>
+#include <sound/sdca_interrupts.h>
+#include <sound/sdca_function.h>
+
+static irqreturn_t sdca_base_irq_handler(int irq, void *data)
+{
+	struct sdca_interrupt *priv = data;
+
+	return IRQ_HANDLED;
+}
+
+#define IRQ_SDCA_OFFSET(reg_number) ((SDW_SCP_SDCA_INT##reg_number) - SDW_SCP_SDCA_INT1)
+#define IRQ_SDCA(number, reg_number) REGMAP_IRQ_REG(number, \
+						    IRQ_SDCA_OFFSET(reg_number), \
+						    SDW_SCP_SDCA_INTMASK_SDCA_##number)
+
+static const struct regmap_irq regmap_irqs[] = {
+	IRQ_SDCA(0, 1),
+	IRQ_SDCA(1, 1),
+	IRQ_SDCA(2, 1),
+	IRQ_SDCA(3, 1),
+	IRQ_SDCA(4, 1),
+	IRQ_SDCA(5, 1),
+	IRQ_SDCA(6, 1),
+	IRQ_SDCA(7, 1),
+	IRQ_SDCA(8, 2),
+	IRQ_SDCA(9, 2),
+	IRQ_SDCA(10, 2),
+	IRQ_SDCA(11, 2),
+	IRQ_SDCA(12, 2),
+	IRQ_SDCA(13, 2),
+	IRQ_SDCA(14, 2),
+	IRQ_SDCA(15, 2),
+	IRQ_SDCA(16, 3),
+	IRQ_SDCA(17, 3),
+	IRQ_SDCA(18, 3),
+	IRQ_SDCA(19, 3),
+	IRQ_SDCA(20, 3),
+	IRQ_SDCA(21, 3),
+	IRQ_SDCA(22, 3),
+	IRQ_SDCA(23, 3),
+	IRQ_SDCA(24, 4),
+	IRQ_SDCA(25, 4),
+	IRQ_SDCA(26, 4),
+	IRQ_SDCA(27, 4),
+	IRQ_SDCA(28, 4),
+	IRQ_SDCA(29, 4),
+	IRQ_SDCA(30, 4),
+};
+
+static const struct regmap_irq_chip sdca_irq_chip = {
+	.name = "sdca_irq",
+
+	.status_base = SDW_SCP_SDCA_INT1,
+	.mask_base = SDW_SCP_SDCA_INTMASK1,
```

Comment:
SDCA IRQs are unmasked by writing a 1 so we should be using unmask_base. Also we need to write a 1 to ack the IRQ so we need to set ack_base = SDW_SCP_SDCA_INT1 as well.